WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/domaintransfer/DashboardCardDomainTransferUtils.kt

```diff
@@ -0,0 +1,149 @@
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.wordpress.android.analytics.AnalyticsTracker
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.DashboardCards
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.DashboardCards.DashboardCard.DomainTransferCardModel
+import org.wordpress.android.ui.mysite.MySiteViewModel
+import org.wordpress.android.ui.mysite.tabs.MySiteTabType
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.util.BuildConfigWrapper
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+import javax.inject.Inject
+import javax.inject.Named
+
+class DashboardCardDomainTransferUtils @Inject constructor(
+    private val appPrefsWrapper: AppPrefsWrapper,
+    private val buildConfigWrapper: BuildConfigWrapper,
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
+) {
+    private var dashboardUpdateDebounceJob: Job? = null
+
+    private val domainTransferCardVisible = AtomicReference<Boolean?>(null)
+    private val waitingToTrack = AtomicBoolean(false)
+    private val currentSite = AtomicReference<Int?>(null)
+
+    fun shouldShowCard(
+        siteModel: SiteModel,
+    ): Boolean {
+        return buildConfigWrapper.isJetpackApp &&
+                !isCardHiddenByUser(siteModel.siteId) &&
+                siteModel.isAdmin &&
+                !siteModel.isWpForTeamsSite
+    }
+
+    fun hideCard(siteId: Long) {
+        appPrefsWrapper.setShouldHideDashboardDomainTransferCard(siteId, true)
+    }
+
+    fun trackCardShown(scope: CoroutineScope, siteSelected: MySiteViewModel.State.SiteSelected?) {
+        // cancel any existing job (debouncing mechanism)
+        dashboardUpdateDebounceJob?.cancel()
+
+        dashboardUpdateDebounceJob = scope.launch(bgDispatcher) {
+            val isVisible = siteSelected
+                ?.dashboardCardsAndItems
+                ?.filterIsInstance<DashboardCards>()
+                ?.firstOrNull()
+                ?.cards
+                ?.any {
+                        card -> card is DomainTransferCardModel
+                } ?: false
+
+            // add a delay (debouncing mechanism)
+            delay(CARD_VISIBLE_DEBOUNCE)
+
+            domainTransferCardVisible.set(isVisible)
+            if (isVisible && waitingToTrack.getAndSet(false)) {
+                trackCardShown(positionIndex(siteSelected))
+            }
+        }.also {
+            it.invokeOnCompletion { cause ->
+                // only set the job to null if it wasn't cancelled since cancellation is part of debouncing
+                if (cause == null) dashboardUpdateDebounceJob = null
+            }
+        }
+    }
```

Comment:
❓ [Question] : Do we need this tracking for the domain transfer card?.

I can see the tracking for domain transfer card already present in `CardsShownTracker.kt`. When I tested the PR i saw two tracking events for the domain card 

🔵 Tracked: my_site_dashboard_card_shown, Properties:{"type":"dashboard_card_domain_transfer","subtype":"dashboard_card_domain_transfer"}
🔵 Tracked: dashboard_card_domain_transfer_shown, Properties: {"position_index":0}

Hence I wondering whether this is needed or not.